tc/sgx/trusted_worker_manager/enclave/CMakeWorkloads.txt

```diff
@@ -0,0 +1,94 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##################################################################
+MACRO(CREATE_SUPPORTED_WORKLOADS_LIST)
+    SET(SUPPORTED_WORKLOADS_LIST "")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "echo-result")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "heart-disease-eval")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "inside-out-eval")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "simple-wallet")
+
+    # A new workload can be added by using below syntax
+    # LIST(SUPPORTED_WORKLOADS_LIST "<workload_id>")
+ENDMACRO()
+
+##################################################################
+MACRO(CREATE_SUPPORTED_WORKLOAD_LIBRARY_LIST)
+    SET(SUPPORTED_WORKLOAD_LIBRARY_LIST "")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "echo")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "heart_disease_eval")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "inside_out_eval")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "simple_wallet")
+
+    # A new workload library can be added by using below syntax
+    # Note: Order of workload id should match the order in
+    #       supported workload library list
+    # LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "<workload_lib_name>")
+ENDMACRO()
+
+##################################################################
+MACRO(CREATE_LINK_WORKLOAD_LIBRARY_LIST)
+    SET(WORKLOAD_LIST "$ENV{WORKLOADS}")
+    SET(WORKLOAD_LIB_LIST "")
+    # Create workload library list based on workloads passed
+    IF("${WORKLOAD_LIST}" STREQUAL "all" OR "${WORKLOAD_LIST} " STREQUAL " ")
+        SET(WORKLOAD_LIB_LIST ${SUPPORTED_WORKLOAD_LIBRARY_LIST})
+    ELSE()
+        FOREACH(workload ${WORKLOAD_LIST})
+            # Check if user provided workload is present in supported workload list
+            # If not found, return -1
+            LIST(FIND SUPPORTED_WORKLOADS_LIST ${workload} index)
+            IF(NOT ${index} EQUAL -1)
+                # Get workload library corresponding to workload identifier
+                LIST(GET SUPPORTED_WORKLOAD_LIBRARY_LIST ${index} workload_lib)
+                LIST(APPEND WORKLOAD_LIB_LIST ${workload_lib})
+            ELSE()
+                MESSAGE(FATAL_ERROR,
+                    "Unable to find workload library corresponding to parsed workload")
+            ENDIF()
+        ENDFOREACH(workload)
+    ENDIF()
+ENDMACRO()
+
+##################################################################
+MACRO(CONFIG_WORKLOAD_LINKER_FLAGS)
+    FOREACH(workload_lib ${WORKLOAD_LIB_LIST})
+        # workload folder name under $TCFHOME/example/apps/<workload_folder>
+        # must match the static library name configured inside workload CMakeLists.txt
+        SET(CMAKE_SHARED_LINKER_FLAGS
+            "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-L,${TCF_TOP_DIR}/examples/apps/build/${workload_lib}/workload")
+    ENDFOREACH(workload_lib)
```

Comment:
Please note that we are hard coding the base directory as `${TCF_TOP_DIR}/examples/apps/`. Let's keep like this for now.  But I would suggest to add a variable at top of this file like `BASE_WORKLOAD_DIRECTORY = ${TCF_TOP_DIR}/examples/apps/` and use it BASE_WORKLOAD_DIRECTORY variable here. as `${BASE_WORKLOAD_DIRECTORY}build/${workload_lib}/workload`

Reply:
Ok

Reply:
Could you please use `BASE_WORKLOAD_DIRECTORY` variable

Reply:
Added BASE_WORKLOAD_DIR to refer TCF_TOP_DIR/examples/apps